src/widgets/reflow.rs

```diff
@@ -189,13 +197,18 @@ where
 
         loop {
             // emit next cached line if present
-            if let Some(line) = self.next_cached_line() {
+            if let Some(line) = self.wrapped_lines.pop_front() {
                 let line_width = line
                     .iter()
                     .map(|grapheme| grapheme.symbol.width() as u16)
                     .sum();
 
-                self.current_line = line;
+                {
+                    let old = mem::replace(&mut self.current_line, line);
+                    if old.capacity() > 0 {
+                        self.pending_line_pool.push(old);
+                    }
+                }
```

Comment:
Why the block here?

Reply:
This is just my coding style. Can remove if you'd like, but I always try to minimally scope variables (in this case `old`).

Reply:
It's not a huge deal, but I'd generally interpret the implication of the block surrounding some small amount of code like this is that old requires some sort of special drop behavior. But without a comment to help guide the next person that reads this, the intent of why that is is lost. For something like this perhaps documenting it with a method name would help (and achieve the minimal scope)? E.g. what do you think `self.replace_current_line(line)` reads better for this block?

Reply:
Sure, `replace_current_line` seems pretty clear. I also renamed the variable `cache` to be clearer.